### PR TITLE
Add CLP and masterKey config options and auditing of _User, _Role, etc

### PR DESCRIPTION
**Pull request by cbaker6**
Awesome package! I was testing and added some more capabilities.

One thing I noticed is that parse-server will only let node modules change class level permissions (CLPs) if the CLPs are already public (which they are by default). It looks like parse-server ignores operations otherwise which I guess is a good thing as pre-installed modules should only be able to use the masterKey and modify CLPs if you want it to.

To circumvent this, you have to point to the index file of parse-audit directly in main.js of your CloudCode. So it looked like the following:

```const ParseAuditor = require('../node_modules/parse-auditor/src/index.js');```

If not changing CLPs or if you don't need the masterKey, using below works:

```const ParseAuditor = require('parse-auditor');```

Note that since this is an auditing package, parse-server admins probably shouldn't use the default CLP's in parse-server as they are public read/write/add fields, etc. and gives all of this access to the auditing classes. So they should set them accordingly and can do so through the config options or Parse Dashboard.

I also fixed a small bug when attempting to audit classes like `_User` and `_Role` when using the default config. The bug occurs because no prefix is set and Parse only lets its own classes begin with `_`

### CODE REVIEW

**Comment by rwblackburn** [comment]
This is a great change @cbaker6  - thank you for contributing it. I'll pull it down and test it locally (unfortunately, I have not had time to add automated testing to this yet).

**Comment by rwblackburn** [comment]
@cbaker6 This catch on making sure the CLP is set properly on the audit classes is a very important one. Especially given that this is an auditing tool, so likely to be used in scenarios where security is a bigger concern. 

I'm thinking it might be worth adding a specific section to the README to explicitly talk about this, maybe even linking to an article like this one to help: https://blog.back4app.com/parse-server-security/

File: src/index.js [hunk 1]
```diff
@@ -30,11 +52,17 @@ const init = (auditModifiedClasses, auditAccessClasses = [], options = {}) => {
...
+    const clp = options.clp || CONFIG_DEFAULTS.clp;
     const config = {
-        classPrefix, classPostfix, fieldPrefix, fieldPostfix, parseSDK,
+        classPrefix, classPostfix, fieldPrefix, fieldPostfix, parseSDK, useMasterKey, clp
     };
 
     auditModifiedClasses.forEach((c) => {
+        if (Object.keys(config.clp).length !== 0){
+            parseSDK.Cloud.define(`audit${config.classPrefix}${c}${config.classPostfix}SetCLP`, async req =>  setCLP(c, config));
+            parseSDK.Cloud.run(`audit${config.classPrefix}${c}${config.classPostfix}SetCLP`);
```
Comment (rwblackburn) [inline src/index.js]: Hi @cbaker6  - Just curious, why the `parseSDK.Cloud.define`/`parseSDK.Cloud.run` combo? Can't we just call `setCLP(c, config)` directly here?

Comment (cbaker6) [reply src/index.js]: Right, I had this left over from when I was initially testing and couldn't get setCLP to fire.

I've tested without define and run and it works, will make the change in a sec

**Comment by cbaker6** [comment]
Below is a default CLP I was using for testing:

```javascript
const auditCLP = {
      get: { requiresAuthentication: true },
      find: { requiresAuthentication: true },
      create: {},
      update: { requiresAuthentication: true },
      delete: { requiresAuthentication: true },
      addField: {},
      protectedFields: {}
    };
```

In this case, will have to pass `useMaterKey: true` through the config for parse-auditor to addFields.

File: src/index.js (hunk 1 above)
Comment (rwblackburn) [reply src/index.js]: Great, thank you

**Comment by rwblackburn** [comment]
The default CLP is helpful, thank you

**Pull request merged**